.github/workflows/cve-daily.yml

```diff
@@ -0,0 +1,373 @@
+
+      # <template: unlink_bin_step>
+      - name: Unlink binary cache
+        id: unlink_bin_step
+        if: always()
+        run: |
+          rm bin
+      # </template: unlink_bin_step>
+    # </template: cve_tests>
+
+  test_cve_report_main:
+    name: Deckhouse images on main
+    needs:
+      - skip_tests_repos
+      - test_cve_base_images_main
+    env:
+      REPO: "dev-registry.deckhouse.io/sys/deckhouse-oss"
+      TAG: "main"
+    # <template: cve_tests>
+    runs-on: [self-hosted, regular]
+    steps:
+
+      # <template: checkout_full_step>
+      - name: Checkout sources
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      # </template: checkout_full_step>
+
+      # <template: login_dev_registry_step>
+      - name: Check dev registry credentials
+        id: check_dev_registry
+        env:
+          HOST: ${{secrets.DECKHOUSE_DEV_REGISTRY_HOST}}
+        run: |
+          if [[ -n $HOST ]]; then echo "::set-output name=has_credentials::true"; fi
+      - name: Login to dev registry
+        uses: docker/login-action@v1.10.0
+        if: ${{ steps.check_dev_registry.outputs.has_credentials == 'true' }}
+        with:
+          registry: ${{ secrets.DECKHOUSE_DEV_REGISTRY_HOST }}
+          username: ${{ secrets.DECKHOUSE_DEV_REGISTRY_USER }}
+          password: ${{ secrets.DECKHOUSE_DEV_REGISTRY_PASSWORD }}
+          logout: false
+      # </template: login_dev_registry_step>
+
+      # <template: login_readonly_registry_step>
+      - name: Check readonly registry credentials
+        id: check_readonly_registry
+        env:
+          HOST: ${{secrets.DECKHOUSE_REGISTRY_READ_HOST}}
+        run: |
+          if [[ -n $HOST ]]; then echo "::set-output name=has_credentials::true"; fi
+      - name: Login to readonly registry
+        uses: docker/login-action@v1.10.0
+        if: ${{ steps.check_readonly_registry.outputs.has_credentials == 'true' }}
+        with:
+          registry: ${{ secrets.DECKHOUSE_REGISTRY_READ_HOST }}
+          username: ${{ secrets.DECKHOUSE_REGISTRY_READ_USER }}
+          password: ${{ secrets.DECKHOUSE_REGISTRY_READ_PASSWORD }}
+          logout: false
+      # </template: login_readonly_registry_step>
+
+      # <template: link_bin_step>
+      - name: Link binary cache
+        id: link_bin_step
+        run: |
+          ln -s ~/deckhouse-bin-cache bin
+      # </template: link_bin_step>
+      - name: Set repo and tag for matrix run
+        if: matrix.tag != ''
+        run: |
+          REPO="registry.deckhouse.io/deckhouse/fe"
+          echo "REPO=${REPO}" >> $GITHUB_ENV
+          TAG=$(echo "${{ matrix.tag }}" | grep -Eo "^\S+")
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+      - name: Run CVE tests
+        run: |
+          echo "⚓️ 📤 Checking out candi/image_versions.yml from ${TAG}..."
+          git checkout --force "${TAG}" -- candi/image_versions.yml
+          echo "⚓️ 🏎 Running CVE tests on ${TAG}..."
+          make cve-report
```

Comment:
Is it correct that this job should not run if previous job is skipped? In this case previous job ('Set repo and tag for matrix run') can return output and this job ('Run CVE tests') may have `if` that checks this output. For example, see steps "Check registry"/"Login to registry".

Reply:
Nope, we shouldn't skip this job simultaneously with a previous one.
Required env vars could be set manually in workflow definition (e.g. for main branch) or via `Set repo and tag for matrix run` step (when executed with matrix strategy).